pkg/webhook/resources/addon/validator.go

```diff
@@ -83,12 +83,35 @@ func validateUpdatedAddon(newAddon *v1beta1.Addon, oldAddon *v1beta1.Addon) erro
 		return werror.NewBadRequest("chart field cannot be changed.")
 	}
 
+	// do not allow change Spec when an operation is onging, but allow change annotation, e.g. adjust the timeout
+	if oldAddon.Status.Status == v1beta1.AddonEnabled && isAddonSpecChanged(newAddon, oldAddon) {
+		return werror.NewBadRequest("addon is being deployed now, please wait and retry")
```

Comment:
did not figure out reason yet, @ibrokethecloud any idea of this scenario?

the `AddonEnabled` should only be a middle status, it will finally be successful or failed (timeout)

@guangbochen , the action `3` seems to be expected, as the `addon` is being deployed now, disable is not allowed.

Reply:
I would assume the current code didn't cover the migration/upgrade logic from the previous addon status; that's why step3 is getting blocked. (also talked with @ibrokethecloud, and he will help to continue on this PR, thanks.)